PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,367 @@
+import random
+from tkinter import *
+
+try:
+    from drawable import *
+    from VisualizationApp import *
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+
+
+class PriorityQueue(VisualizationApp):
+    CELL_SIZE = 50
+    CELL_BORDER = 2
+    CELL_BORDER_COLOR = 'black'
+    ARRAY_X0 = 100
+    ARRAY_Y0 = 100
+
+    nextColor = 0
+
+    def __init__(self, size=12, title="Priority Queue", **kwargs):
+        super().__init__(title=title, **kwargs)
+        self.size = size
+        self.title = title
+        self.list = [None]*self.size
+        self.nItems = 0
+        self.buttons = self.makeButtons()
+        self.index = self.createIndex(self.nItems-1, 'front', level=-1) # indicate priority
+        
```

Comment:
Don't create the `nItems` index here.  It's better to have it in the `display()` method so it gets recreated when New is run.